### ISSUE TITLE
Questions regarding usage of navigate() and navigateLink() [Navigation React Native]

### ISSUE THREAD

**Issue by arunmenon1975** [grahammendick/navigation#278]
tldr;

1) welcome => menu => welcome (doesn't work with `navigate` but works with `navigateLink`, `navigateBack` and hardware back press)

2) welcome => menu [pass initial nav props] => menu [pass changed nav props calling the same state key]  (can it work)

My (simplified) app has this flow: Open into a `Welcome` scene and then proceed to an inner scene (with tabs, but it does not matter here since my doubts remain intact even without it), say a `Menu` scene. 

I use `stateNavigator.navigate('menu')` from my welcome scene and i reach the menu scene. I understand that `Navigator` will thus have 'dropped' crumbs at  these scenes. But when i explicitly navigate from my `Menu` scene to the `Welcome` scene (also equivalent to a back button press) using `stateNavigator.navigate('welcome')` the view doesn't seem to change but i see from simple `console.log` statements that the Welcome component has indeed been triggered. [I can of course easily `navigateBack(1)` but i may have a usecase somewhere else where i use the unique key to reach a particular scene regardless of whether it has been visited or not]

The view changes correctly when i use `stateNavigator.navigateLink('welcome')` however.

So my questions are:

1) Why doesn't the `navigate` work as expected (the view is not changing, but i see it gets triggered however by using log statement) while `navigateLink` does for navigating to a previously visited scene. I understand `navigateLink` is especially useful for deep linking in the native side of things. 
2) From within a scene, can i navigate to the same scene but pass different props. Eg. In the Menu scene, i am actually using `ViewPagerAndroid` as a tab equivalent and though i use state/redux for setting/getting the active tab, i wanted to use Navigation state props to pass data and ensure that the correct tab is active (eg:  from any tab pass  `stateNavigator.navigate('menu',{activeTab:"the-relevant-tab-id"})` and use the changed props to effect the activation/deactivation). Currently i get the '..trying to remove a view index above child count 0...' error when i attempt `stateNavigator.navigateLink('menu',{activeTab: "A Value here"})` and when i attempt a `stateNavigator.navigate('menu',{activeTab: "A Value here"})` nothing happens.

Please see the simplified code below:

App.js

```
import React from 'react';
import {Alert, BackHandler, Text} from 'react-native';
import {StateNavigator} from 'navigation';
import {NavigationHandler} from 'navigation-react';
import {NavigationStack} from 'navigation-react-native';
import {Provider} from 'react-redux';

import store from './src/redux/store';

import MenuScene from './src/scenes/Menu/Menu';
import Welcome from './src/scenes/Welcome/Welcome';
//import ConnectedIntlProvider from './src/helpers/wrappers/ConnectedIntlProvider';
//import ConnectedAppInit from './src/helpers/wrappers/ConnectedAppInit';
import { MenuProvider } from 'react-native-popup-menu';

const stateNavigator = new StateNavigator([
  {
    key: 'welcome',
    trackCrumbTrail: true
  }, {
    key: 'menu',
    trackCrumbTrail: true
  }
]);
const {welcome, menu} = stateNavigator.states;
welcome.renderScene = () => <Welcome/>;
menu.renderScene = () =>  <MenuScene/> ;
stateNavigator.navigate('welcome');
class App extends React.Component {

  render() {
    return (
        <Provider store={store}>
                <NavigationHandler stateNavigator={stateNavigator}>
                  <NavigationStack  />
                </NavigationHandler>
        </Provider>
    );
  }
}
export default App;

```

Welcome.js

```
import React, { Component } from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import WelcomeCmp from '../../components/Welcome/Welcome';


class Welcome extends Component {
  render() {
      return (
        <SafeAreaView style={styles.container}>
        
        <WelcomeCmp  />
      </SafeAreaView>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#E91E63',
    position: "relative",
  },
  

});

export default Welcome;
```

WelcomeCmp.js

```
import { NavigationContext } from 'navigation-react';
import React from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';
//import {gotoScreen, routesKeys} from '../../routes';

const WelcomeCmp = (props) => {
    console.log("The Welcome component");
    return <NavigationContext.Consumer>
    {({stateNavigator}) => {
        console.log(stateNavigator);
      return (<View style={{flex: 1,}}>
    <Text>
        Welcome
    </Text>
    <Button
        title="To Menu Screen"
        onPress={()=>stateNavigator.navigate("menu")}
       />
   
</View>
      )}
    }
    </NavigationContext.Consumer>
}


export default WelcomeCmp;

```

Tabs.Android.js (simplified)

```
import { NavigationContext } from 'navigation-react';
import React, { PureComponent } from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
//import { routesKeys } from '../../routes/';

const DailySpecialsCmp = (props)=>{
  return <NavigationContext.Consumer>
  {({stateNavigator}) => {
    return (<View style={{flex:1,justifyContent: 'center',   alignItems: 'center', backgroundColor: '#C2185B',  }}>
  <Text>
    Daily Specials
  </Text>
  <TouchableOpacity 
    onPress={() => { 
      stateNavigator.navigate('menu',{activeTab: "A Value here"})
    }}
  >
    <Text> An Inner Scene</Text>
  </TouchableOpacity>
  
</View>
  )}
}
  </NavigationContext.Consumer>
}

class Tabs extends PureComponent {
    render(){
        return( 
           <DailySpecialsCmp /> 
        );
    }
}

export default Tabs;

```

**Comment by grahammendick** [comment]
Sounds like #275. Upgrade to v5.4.1 and all will be well

**Comment by arunmenon1975** [comment]
Confirmed that upgrading to the latest version fixes the issue.
